src/scripts/main.js

```diff
@@ -1,3 +1,134 @@
 'use strict';
 
-// write code here
+function appendRow(button) {
+  const table = document.querySelector('.field > tbody');
+  const existRow = table.querySelector('tr');
+  const newRow = document.createElement('tr');
+
+  if (table.querySelectorAll('tr').length >= 10) {
+    button.disabled = true;
+
+    return 0;
+  }
+
+  newRow.innerHTML = existRow.innerHTML;
+
+  table.appendChild(newRow);
+
+  toggleButtons();
+}
+
+function removeRow(button) {
+  const table = document.querySelector('.field > tbody');
+  const rowToDel = table.querySelector('tr');
+
+  if (table.querySelectorAll('tr').length <= 2) {
+    button.disabled = true;
+
+    return 0;
+  }
+
+  rowToDel.remove();
+
+  toggleButtons();
+}
+
+function appendCol(button) {
+  const table = document.querySelector('.field > tbody');
+  const rows = [...table.querySelectorAll('tr')];
+
+  if (rows[0].querySelectorAll('td').length >= 10) {
+    button.disabled = true;
+
+    return 0;
+  }
+
+  rows.forEach((row) => {
+    const newCell = document.createElement('td');
+
+    row.appendChild(newCell);
+  });
+
+  toggleButtons();
+}
+
+function removeCol(button) {
+  const table = document.querySelector('.field > tbody');
+  const rows = [...table.querySelectorAll('tr')];
+
+  if (rows[0].querySelectorAll('td').length <= 2) {
+    button.disabled = true;
+
+    return 0;
+  }
+
+  rows.forEach((row) => {
+    const cellToDel = row.querySelector('td');
+
+    cellToDel.remove();
+  });
+
+  toggleButtons();
+}
+
+function toggleButtons() {
+  const table = document.querySelector('.field > tbody');
+  const rows = table.querySelectorAll('tr');
+  const rowCount = rows.length;
+  const cols = rows[0] ? rows[0].querySelectorAll('td') : [];
```

Comment:
In the `toggleButtons` function, if the table is empty, `rows[0]` will be undefined, which could lead to errors when trying to access `rows[0].querySelectorAll('td')`. Consider adding a check to ensure `rows[0]` exists before accessing its properties.